REST & API: decorate some APIs with session managers. Closes #5002

rcarpa opened the pull request:
Only create the session object once per api call to rules enpoint
and pass it through. The goal is to test if the number of un-needed
rollbacks on the database side is decreased thanks to avoiding
sessionmaker.remove() calls.

<!-- Please read https://github.com/rucio/rucio/blob/master/CONTRIBUTING.rst before submitting a pull request -->

bari12 commented on lib/rucio/api/rule.py:
```diff
@@ -160,102 +168,113 @@ def list_replication_rules(filters={}, vo='def'):
     """
     kwargs = {'rule_id': rule_id}
-    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs):
+    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs, session=session):
```
`is_multi_vo` asks for `config_get_bool` which might query config table. Needs to be added there (Or we block it from querying the config table via the method option)

bari12 commented on lib/rucio/api/rule.py:
```diff
@@ -160,102 +168,113 @@ def list_replication_rules(filters={}, vo='def'):
     """
     kwargs = {'rule_id': rule_id, 'options': options}
-    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs):
+    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs, session=session):
```
See above

bari12 commented on lib/rucio/api/rule.py:
```diff
@@ -264,36 +283,40 @@ def reduce_replication_rule(rule_id, copies, exclude_expression, issuer, vo='def
     """
     kwargs = {'rule_id': rule_id, 'copies': copies, 'exclude_expression': exclude_expression}
-    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs):
+    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs, session=session):
```
See above

bari12 commented on lib/rucio/api/rule.py:
```diff
@@ -264,36 +283,40 @@ def reduce_replication_rule(rule_id, copies, exclude_expression, issuer, vo='def
     """
     kwargs = {'rule_id': rule_id}
-    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs):
+    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs, session=session):
```
See above

bari12 commented on lib/rucio/api/rule.py:
```diff
@@ -311,12 +334,13 @@ def move_replication_rule(rule_id, rse_expression, activity, source_replica_expr
         'source_replica_expression': source_replica_expression
     }
-    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs):
+    if is_multi_vo() and not has_permission(issuer=issuer, vo=vo, action='access_rule_vo', kwargs=kwargs, session=session):
```
See above

Commit:
REST & API: decorate api/rule with session managers. #5002

Only create the session object once per api call to rules enpoint
and pass it through. The goal is to test if the number of un-needed
rollbacks on the database side is decreased thanks to avoiding
sessionmaker.remove() calls.

Commit:
REST & API: decorate api/replica with session managers. #5002

Commit:
REST & API: decorate api/did with session managers. #5002
